kim/field.py

```diff
@@ -0,0 +1,104 @@
+# kim/field.py
+# Copyright (C) 2014-2015 the Kim authors and contributors
+# <see AUTHORS file>
+#
+# This module is part of Kim and is released under
+# the MIT License: http://www.opensource.org/licenses/mit-license.php
+
+from .utils import set_creation_order
+from .pipelines import Input, Output, StringInput, StringOutput
+
+
+class FieldError(Exception):
+    pass
+
+
+class FieldOpts(object):
+
+    def __init__(self, **opts):
+
+        self._opts = opts
+
+        self.required = opts.pop('required', False)
+        self.default = opts.pop('default', None)
+        self.source = opts.pop('source', None)
+        self.name = opts.pop('name', self.source)
```

Comment:
Mapper metaclass should set attribute_name as name it was declared on the Mapper as, then:
```python
self.name = opts.pop('name', self.attribute_name)
self.source = opts.pop('source', self.name)
```